addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8800,6 +8916,78 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
+     * of the previous if the shape flip on Y.
+     *
+     * @private
+     * @param {String} shape identifier of the selected shape.
+     * @param {Element} $target Element for which the color has to be computed.
+     */
+    _getConnectionsColors(shape, $target = this.$target[0]) {
+        if (!shape.includes("web_editor/Connections/")) {
+            return {};
+        }
+        const defaultColor = this._getShapeDefaultColors(shape);
+        const defaultKey = Object.keys(defaultColor);
+        let nextElement = undefined;
+        const isFlippedY = this._getShapeData($target).flip.includes("y");
+        const isInFooter = $target.closest("footer");
+        const hasPreviousSibling = !!$target.previousElementSibling;
+
+        if (isFlippedY) {
+            nextElement = isInFooter
+                ? hasPreviousSibling
+                    ? this.ownerDocument.body.querySelector("footer")
+                    : this.ownerDocument.body.querySelector("main > #wrap > section:last-child")
+                : $target.previousElementSibling || this.ownerDocument.body.querySelector("nav");
+        } else {
+            nextElement = isInFooter
+                ? this.ownerDocument.body.querySelector("footer")
+                : $target.nextElementSibling || this.ownerDocument.body.querySelector("footer");
+        }
+        const nextElementBgColor = getComputedStyle(nextElement).backgroundColor;
+        const nextElementHexcolor = rgbToHex(nextElementBgColor);
+        const curBgColor = isInFooter
+            ? getComputedStyle(this.ownerDocument.body.querySelector("footer")).backgroundColor
+            : getComputedStyle($target).backgroundColor;
+        let color = undefined;
+        // We compare the HexColor to avoid comparing rgb and rgba together.
+        const curBgHexColor = rgbToHex(curBgColor);
+        if (curBgHexColor != nextElementHexcolor) {
+            color = Object.assign(defaultColor, { [defaultKey[0]]: nextElementHexcolor });
+        } else {
+            color = Object.assign(defaultColor, {
+                [defaultKey[0]]: this._getContrastingColor(nextElementHexcolor),
+            });
+        }
+        return color;
+    },
+    /**
+     * Compute the luminance of a color.
+     *
+     * @private
+     */
+    _getLuminance(hexColor) {
+        const r = parseInt(hexColor.substr(1, 2), 16);
+        const g = parseInt(hexColor.substr(3, 2), 16);
+        const b = parseInt(hexColor.substr(5, 2), 16);
+        return 0.2126 * r + 0.7152 * g + 0.0722 * b;
+    },
+    /**
+     * Return the whitish or the blackish color of the current theme
+     * based on the luminance of the given color.
+     *
+     * @private
+     */
+    _getContrastingColor(color) {
+        const luminance = this._getLuminance(color);
+        return luminance > 128
+            ? weUtils.getCSSVariableValue("o-color-5")
+            : weUtils.getCSSVariableValue("o-color-4");
```

Comment:
- Maybe use the name in the URL directly instead of its actual color value
- But maybe check if in a dark theme, you don't have to compare color 4 & 5 first